test/apps/iodemo/io_demo.cc

```diff
@@ -887,12 +943,14 @@ class DemoServer : public P2pDemoCommon {
         VERBOSE_LOG << "sending IO read data";
         assert(opts().max_data_size >= msg->data_size);
 
-        BufferIov *iov           = _data_buffers_pool.get();
-        SendCompleteCallback *cb = _send_callback_pool.get();
+        BufferIov *iov            = _data_buffers_pool.get();
+        SendCompleteCallback *cb  = _send_callback_pool.get();
+        ConnectionStat &conn_stat = _conn_stat_map.find(conn->id())->second;
```

Comment:
maybe just _conn_stat_map[conn->id()] ?

Reply:
it more bug-friendly since operator[] inserts an elem if key is not present, find returns end() iterator

Reply:
so end->second will throw exception?

Reply:
I'm not sure if it's guaranteed by standard, but STL documentation says that end() iterator cannot be dereferenced.